src/components/withCurrentReportID.tsx

```diff
@@ -39,7 +40,18 @@ function CurrentReportIDContextProvider(props: CurrentReportIDContextProviderPro
      */
     const updateCurrentReportID = useCallback(
         (state: NavigationState) => {
-            setCurrentReportID(Navigation.getTopmostReportId(state) ?? '');
+            const reportID = Navigation.getTopmostReportId(state) ?? '';
+
+            /**
+             * This is to make sure we don't set the undefined as reportID when
+             * switching between chat list and settings->workspaces tab.
+             * and doing so avoid unnecessary re-render of `useReportIDs`.
+             */
+            const params = state.routes[state.index].params;
```

Comment:
Is this null free? `state.routes[state.index]` might not exist for some reason.
I am concerned here because I experienced a few crashes in the past came from this assertion.

Reply:
Good point, I also remember some crashes stemming from this